File: app/src/main/java/uk/co/ribot/androidboilerplate/ui/main/MainMvpView.java
package uk.co.ribot.androidboilerplate.ui.main;

import com.sunny.commonbusiness.base.MvpView;

import java.util.List;

import uk.co.ribot.androidboilerplate.data.model.bean.Subject;

public interface MainMvpView extends MvpView {

    void showSubjects(List<Subject> subjects);

    void showSubjectsEmpty();

    void showError();

}

File: app/src/main/java/uk/co/ribot/androidboilerplate/ui/main/MainMvpView.kt
package uk.co.ribot.androidboilerplate.ui.main

import uk.co.ribot.androidboilerplate.data.model.bean.Subject
import uk.co.ribot.androidboilerplate.ui.base.MvpView

interface MainMvpView : MvpView {

    fun showSubjects(subjects: List<Subject>)
    fun showSubjectsEmpty()
    fun showError()

}
